core/src/main/java/hudson/TcpSlaveAgentListener.java

```diff
@@ -135,9 +136,11 @@ public String getAdvertisedHost() {
           return CLI_HOST_NAME;
         }
         try {
-            return new URL(Jenkins.get().getRootUrl()).getHost();
+            return new URI(Jenkins.get().getRootUrl()).toURL().getHost();
         } catch (MalformedURLException e) {
             throw new IllegalStateException("Could not get TcpSlaveAgentListener host name", e);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
```

Comment:
does not help diagnose an ussue? - add a message like above?
```suggestion
             throw new IllegalStateException("Could not get TcpSlaveAgentListener host name", e);
```

Reply:
I applied 
            `throw new URISyntaxException(e.getReason(), e.getInput());`